### PR TITLE
feat: validate maxTotalBalance

### PR DESCRIPTION
**Pull request by CarmitKl**
The total value locked (TVL) in the bridge on L1 is limited for each token type: currently, the limit is $100k for each token.
So this PR adds the validation to prevent executing a transaction if the selectetToken will pass its 'maxTotalBalance' value.
If so- the user will see the error-modal with an appropriate explanation.

- [ ] Changes have been done against master branch, and PR does not conflict
- [ ] New unit / functional tests have been added (whenever applicable)
- [ ] Test are passing in local environment
- [ ] Docs have been updated
- [ ] PR title is follow the [standard-version](https://www.conventionalcommits.org/) convention: `<type>(optional subject): <description>`, e.g: `fix: minor typos in code`

---
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starkgate-frontend/101)

### CODE REVIEW

**Comment by dan-ziv** [comment]
@CarmitKl please change the PR title to 
feat: validate maxTotalBalance

File: src/hooks/useTransfer.js [hunk 1]
```diff
@@ -41,10 +48,27 @@ export const useTransferToL2 = () => {
...
         const depositHandler = isEthToken ? depositEth : deposit;
+        const tokenContract = isEthToken && getTokenContract(tokenAddress);
         logger.log('Prepared contract and handler', {isEthToken, bridgeContract, depositHandler});
+
+        const isReachedMaxTotalBalance = async () => {
+          const tokenMaxTotalBalance = await maxTotalBalance({decimals, contract: bridgeContract});
+          const tokenBalance = isEthToken
+            ? await ethBalanceOf(bridgeAddress[chainId])
+            : await balanceOf({
+                account: tokenContract,
```
Comment (dan-ziv) [inline src/hooks/useTransfer.js:59]: `account` is an address.
It should be `bridgeAddress[chainId]`

File: src/hooks/useTransfer.js [hunk 2]
```diff
@@ -41,10 +48,27 @@ export const useTransferToL2 = () => {
...
+                decimals,
+                contract: getL1TokenContract(tokenAddress)
```
Comment (dan-ziv) [inline src/hooks/useTransfer.js:61]: Why not just `tokenContract`?

File: src/hooks/useTransfer.js [hunk 3]
```diff
@@ -41,10 +48,27 @@ export const useTransferToL2 = () => {
...
+              });
+          return tokenMaxTotalBalance < amount + tokenBalance;
+        };
+        if (isReachedMaxTotalBalance()) {
```
Comment (dan-ziv) [inline src/hooks/useTransfer.js:65]: Add space

File: src/hooks/useTransfer.js [hunk 4]
```diff
@@ -41,10 +48,27 @@ export const useTransferToL2 = () => {
         const isEthToken = utils.token.isEth(symbol);
         const bridgeContract = getTokenBridgeContract(bridgeAddress);
         const depositHandler = isEthToken ? depositEth : deposit;
+        const tokenContract = isEthToken && getTokenContract(tokenAddress);
```
Comment (dan-ziv) [inline src/hooks/useTransfer.js:51]: why the && ?

**Comment by dan-ziv** [comment]
@CarmitKl Please add `useMaxTotalBalance` hook from which you will call the API and cache the response.

File: src/hooks/useTransfer.js (hunk 4 above)
Comment (CarmitKl) [reply src/hooks/useTransfer.js:51]: this line was moved up from line 47. this variable is relevant only if **isEthToken** is true, and this is the shortest way to write it.

Comment (dan-ziv) [reply src/hooks/useTransfer.js:51]: Why do u need this statement?

File: src/hooks/useTransfer.js (hunk 2 above)
Comment (CarmitKl) [reply src/hooks/useTransfer.js:61]: the function definition asks specifically arguments with names: account, decimals, contract

Comment (dan-ziv) [reply src/hooks/useTransfer.js:61]: Again:
Why not just `contract: tokenContract`

**Pull request merged**